src/EventStore.Core/Services/Transport/Http/Authentication/ClientCertificateAuthenticationProvider.cs

```diff
@@ -20,25 +21,34 @@ public class ClientCertificateAuthenticationProvider : IHttpAuthenticationProvid
 			if (clientCertificate is null) return false;
 
 			bool hasReservedNodeCN;
+			string clientCertificateCN;
 			try {
-				hasReservedNodeCN = clientCertificate.GetNameInfo(X509NameType.SimpleName, false) == _certificateReservedNodeCommonName;
+				clientCertificateCN = clientCertificate.GetCommonName();
+				hasReservedNodeCN = clientCertificateCN == _certificateReservedNodeCommonName;
 			} catch (CryptographicException) {
 				return false;
 			} catch (NullReferenceException) {
 				return false;
 			}
 
+			if (!hasReservedNodeCN) {
+				var ip = context.Connection.RemoteIpAddress?.ToString() ?? "<unknown>";
+				Log.Error(
+					"Connection from node: {ip} was denied because it's CN: {clientCertificateCN} does not match with the reserved node CN: {reservedNodeCN}",
+					ip, clientCertificateCN, _certificateReservedNodeCommonName);
+				return false;
+			}
+
 			bool hasIpOrDnsSan = clientCertificate.GetSubjectAlternativeNames()
 				.Where(x => x.type is CertificateNameType.DnsName or CertificateNameType.IpAddress)
 				.IsNotEmpty();
 
-			if (hasReservedNodeCN && hasIpOrDnsSan) {
-				request = new HttpAuthenticationRequest(context, "system", "");
-				request.Authenticated(SystemAccounts.System);
-				return true;
-			}
-
-			return false;
+			if (!hasIpOrDnsSan)
+				return false;
```

Comment:
```suggestion
			if (!hasIpOrDnsSan) {
				Log.Error("Connection from node: {ip} was denied because its certificate does not have any IP or DNS Subject Alternative Names (SAN).", ip);
				return false;
			}
```

suggestion: we can also log an error when there is no IP or DNS SAN